cpp/src/bitmask/null_mask.cu

```diff
@@ -695,33 +617,46 @@ rmm::device_buffer bitmask_binop(Binop op,
   }
 
   if (masks.size() > 0) {
-    return cudf::detail::bitmask_binop(op, masks, offsets, view.num_rows(), stream, mr);
+    return cudf::detail::bitmask_binop(
+      [] __device__(bitmask_type left, bitmask_type right) { return left & right; },
+      masks,
+      offsets,
+      view.num_rows(),
+      stream,
+      mr);
   }
 
   return null_mask;
 }
 
-// Returns the bitwise AND of the null masks of all columns in the table view
-rmm::device_buffer bitmask_and(table_view const &view,
-                               rmm::cuda_stream_view stream,
-                               rmm::mr::device_memory_resource *mr)
-{
-  return bitmask_binop(
-    [] __device__(bitmask_type left, bitmask_type right) { return left & right; },
-    view,
-    stream,
-    mr);
-}
-
 rmm::device_buffer bitmask_or(table_view const &view,
                               rmm::cuda_stream_view stream,
                               rmm::mr::device_memory_resource *mr)
 {
-  return bitmask_binop(
-    [] __device__(bitmask_type left, bitmask_type right) { return left | right; },
-    view,
-    stream,
-    mr);
+  CUDF_FUNC_RANGE();
+  rmm::device_buffer null_mask{0, stream, mr};
+  if (view.num_rows() == 0 or view.num_columns() == 0) { return null_mask; }
+
+  std::vector<bitmask_type const *> masks;
+  std::vector<size_type> offsets;
+  for (auto &&col : view) {
+    if (col.nullable()) {
+      masks.push_back(col.null_mask());
+      offsets.push_back(col.offset());
+    }
+  }
+
+  if (static_cast<size_type>(masks.size()) == view.num_columns()) {
```

Comment:
If all of the columns in the table aren't nullable, then you'll be returning an uninitialized `null_mask`.

Reply:
Yes, following logic of the bitmask_and functionality for bitwise_or. If any column has no nulls, it does not matter what are in the other columns and the resulting bitmask should represent a column with no nulls.

Reply:
Ah, I didn't realize `bitmask_and` returned an empty buffer if there aren't any nullable columns. I thought it returned a buffer of all the bits set to 1. Makes sense.